src/main/java/com/mael/ttt/players/HumanPlayer.java

```diff
@@ -23,24 +23,24 @@ public Mark getMark() {
         return mark;
     }
 
-    private String returnValidCellIndex() {
-        String cellIndex      = gameUI.getInput();
-        String validCellIndex = "[1-9]";
-
-        while (!cellIndex.matches(validCellIndex)) {
+    private String returnValidCellIndex(Board board) {
+        String cellIndex = gameUI.getInput();
+        while (isNotValidCellIndex(board, cellIndex)) {
             gameUI.printNotValidCellMessage();
             cellIndex = gameUI.getInput();
         }
-
         return cellIndex;
     }
 
+    private boolean isNotValidCellIndex(Board board, String cellIndex) {
+        return !gameUI.getBoardIndexesAsStrings(board.getSize()).contains(cellIndex);
+    }
+
     private String returnEmptyCellIndex(Board board, String cellIndex) {
```

Comment:
I'd say yes - minus the string business, since strings are not used with a `Board`.
You're right, the user interface is responsible to convert between strings from a user and "real indexes" (read: `int`s) that are needed in order to be able to work with an instance of `Board`.
But that should be it. Nothing more. Just translating `"1" -> 1`.
Then the board can handle whatever input is given and act appropriately. Like rejecting a move because it's out of bounds, or the cell is already occupied, etc.